Core: updated creature type flags enum

ForesterDev opened the pull request:
**Changes proposed**:
- Updated according https://trinitycore.atlassian.net/wiki/display/tc/creature_template#creature_template-type_flags

**Target branch(es)**: 335/6x

Commit:
Update CreatureTypeFlags enum according to TrinityCoreWiki

https://trinitycore.atlassian.net/wiki/display/tc/creature_template#creature_template-type_flags

Shauren commented:
You can add even more names, https://github.com/TrinityCore/WowPacketParser/blob/master/WowPacketParser/Enums/CreatureTypeFlag.cs

Commit:
define more unknown creature type_flags and correct some known names

Source: https://github.com/TrinityCore/WowPacketParser/blob/master/WowPacketParser/Enums/CreatureTypeFlag.cs

Commit:
remove unexpected change

Commit:
correct naming

Treeston commented on src/server/game/Miscellaneous/SharedDefines.h:
```diff
@@ -2576,38 +2576,38 @@ enum CreatureFamily
+    CREATURE_TYPE_FLAG_EXOTIC_PET                           = 0x00010000,   // Can be tamed by hunter as exotic pet
+    CREATURE_TYPE_FLAG_USE_DEFAULT_COLLISION_BOX            = 0x00020000,   // Collision related. (always using default collision box?)
+    CREATURE_TYPE_FLAG_IS_SIEGE_WEAPON                      = 0x00040000,
+    CREATURE_TYPE_FLAG_DOES_NOT_COLLIDE_WITH_MISSILES       = 0x00080000,   // Projectiles can collide with this creature - interacts with TARGET_DEST_TRAJ
```
Projectiles _can_, or _cannot_? Comment says "can", but name says `DOES_NOT_COLLIDE`?

joschiwald replied:
implementation says "can"
creatures does not collide with missiles by default

Shauren replied:
yeah, name is wrong in wpp it seems (todo: open that old legendary 13850 cata debug build that was the source of this)

ForesterDev replied:
Should we rename this flag to CREATURE_TYPE_FLAG_CAN_COLLIDE_WITH_MISSILES according the implementation in TrinityCore, despite the name in WPP?

Shauren replied:
yes

joschiwald commented on src/server/game/Miscellaneous/SharedDefines.h:
```diff
@@ -2576,38 +2576,38 @@ enum CreatureFamily
     CREATURE_TYPEFLAGS_IS_SIEGE_WEAPON                      = 0x00040000,
-    CREATURE_TYPEFLAGS_PROJECTILE_COLLISION                 = 0x00080000,   // Projectiles can collide with this creature - interacts with TARGET_DEST_TRAJ
-    CREATURE_TYPEFLAGS_HIDE_NAMEPLATE                       = 0x00100000,
+    CREATURE_TYPEFLAGS_DOES_NOT_COLLIDE_WITH_MISSILES       = 0x00080000,   // Projectiles can collide with this creature - interacts with TARGET_DEST_TRAJ
```
that sounds like the opposite

Commit:
Rename CREATURE_TYPE_FLAG_DOES_NOT_COLLIDE_WITH_MISSILES to CREATURE_TYPE_FLAG_CAN_COLLIDE_WITH_MISSILES

Aokromes commented:
Any reason to don't merge this?
